internal/controllers/watch/pruning.go

```diff
@@ -0,0 +1,48 @@
+package watch
+
+import (
+	"context"
+	"fmt"
+
+	apiv1 "github.com/Azure/eno/api/v1"
+	"github.com/go-logr/logr"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type pruningController struct {
+	client client.Client
+}
+
+func (c *pruningController) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	comp := &apiv1.Composition{}
+	err := c.client.Get(ctx, req.NamespacedName, comp)
+	if err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	for i, ir := range comp.Status.InputRevisions {
+		if hasBindingKey(comp, ir.Key) {
```

Comment:
Just to be clear, there's always one input revision per input and this only prunes when the composition stops binding the input right?

Reply:
Exactly